lib/reverse_sentence.rb

```diff
@@ -1,6 +1,38 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^3), cubic, because there are three while loops within reverse_sentence() and reverse() that are nested. The amount of iterations may increase proportionately to the data being passed through.
+# Space complexity: O(1), constant, because the only data that is being stored is the start/ending/temp variables, and the array being passed through is being reversed in place.
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  return [] if my_sentence.nil? || my_sentence.length == 0
+  
+  reverse(my_sentence, 0, my_sentence.length-1)
+  
+  start = 0
+  
+  while start < my_sentence.length
```

Comment:
This loop is also O(n), but note it's not nested with the reverse_sententence method.